app/src/main/java/de/westnordost/streetcomplete/screens/main/map/QuestPinsManager.kt

```diff
@@ -63,8 +63,10 @@ class QuestPinsManager(
 
     private val visibleQuestsListener = object : VisibleQuestsSource.Listener {
         override fun onUpdatedVisibleQuests(added: Collection<Quest>, removed: Collection<QuestKey>) {
-            pinsUpdateJob?.cancel()
-            pinsUpdateJob = viewLifecycleScope.launch { updateQuestPins(added, removed) }
+            pinsUpdateJob = viewLifecycleScope.launch {
+                pinsUpdateJob?.join() // don't cancel, as updateQuestPins only updates existing data
```

Comment:
I think this does not make sense. At the point where what is in pinsUpdateJob is executed, pinsUpdateJob is already `this` job. What happens if a job joins on itself?

Reply:
Oh, right... it should be using `oldPinsUpdateJob`.

> What happens if a job joins on itself?

Don't know, but I had unintentionally tried `cancelAndJoin` in on the job itself, and it was just canceled and nothing else. So I guess in "best" case nothing happens.